Fix C4459: Rename a function parameter `profiler_manager` to avoid hiding the global declaration.

fsb4000 opened the pull request:
Hello.
Thank you for your library!
I tried to upgrade from 1.8.5 to 1.9.0 and I faced with an error:
```
D:\a\_work\1\s\benchmarks\google-benchmark\src\benchmark_runner.cc(129,35): error C2220: the following warning is treated as an error
                 ProfilerManager* profiler_manager) {
                                  ^
D:\a\_work\1\s\benchmarks\google-benchmark\src\benchmark_runner.cc(129,35): warning C4459: declaration of 'profiler_manager' hides global declaration
D:\a\_work\1\s\benchmarks\google-benchmark\src\benchmark_runner.cc(65,18): note: see declaration of 'benchmark::internal::profiler_manager'
ProfilerManager* profiler_manager = nullptr;
                 ^
```
Would you mind a slight renaming to solve the issue?
Feel free to suggest a better name...

LebedevRI commented:
Looks like MSVC CI jobs are not in warnings-as-error mode,
so if you care about that compiler, i'd suggest to also
adjust the CI infra here so new warnings don't go unnoticed,
and make the build green. Does that sound like too much to ask for?

fsb4000 commented:
Sure, I will modify the CI.

LebedevRI commented:
Thank you!

fsb4000 commented:
I restored my changes and I added only `-WX` flag.
I did:
```
cmake -E make_directory "build"
cmake -E chdir "build" cmake -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DCMAKE_BUILD_TYPE=Release ../
cmake --build "build" --config Release
```
and I got:
![изображение](https://github.com/user-attachments/assets/709708e6-17f8-4bd0-ab87-228f01b0d91b)
After that I added my renaming and tried the build again. It passed.
Let's look if the CI will pass.

LebedevRI commented:
(#1838 suggests it should not)

fsb4000 commented:
Oh, yes. I forgot that I locally removed `BENCHMARK_INTERNAL_CACHELINE_ALIGNED` because it was intentionally and not a bug, I would change the MS STL CI to not warn on it.

fsb4000 commented:
`Warning	C26495` is a code analyzer warning. We should pass additional flag `/analyze` to compiler to activate it: https://learn.microsoft.com/en-us/cpp/build/reference/analyze-code-analysis?view=msvc-170
https://learn.microsoft.com/en-us/cpp/code-quality/c26495?view=msvc-170
https://learn.microsoft.com/en-us/cpp/code-quality/?view=msvc-170

LebedevRI reviewed (approved):
Looks like a non-intrusive improvement to me.

LebedevRI commented:
Hm, akctually, looking at `include/benchmark/benchmark.h`, there's precedent for more targeted warning hiding:
https://github.com/google/benchmark/blob/6126d2a2052bb48d3472ac0468ade50397d393c5/include/benchmark/benchmark.h#L332-L336
I looked for that before, but didn't find it on the first try...
I think we might want to do the same here.

fsb4000 commented:
Sure.

And about code analysis.

I got
```
C:\Dev\STL\benchmarks\google-benchmark\src\timers.cc(282,1): error C2220: следующее предупреждение рассматривается как ошибка [C:\Dev\STL\benchmarks\google-benchmark\build\src\benchmark.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\src\timers.cc(280): warning C6053: Предыдущий вызов "strncpy" может привести к тому, что строка "tz_offset" не будет завершаться нулем.: Lines: 211, 212, 213, 214, 216, 217, 218, 219, 231, 233, 236
, 243, 245, 266, 271, 274, 276, 278, 280 [C:\Dev\STL\benchmarks\google-benchmark\build\src\benchmark.vcxproj]
  Создание кода...
  gtest-all.cc
  gmock-all.cc
  gmock_main.cc
  Выполнение Code Analysis для C/C++...
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest/internal/gtest-internal.h(627,1): error C2220: следующее предупреждение рассматривается как ошибка [C:\Dev\STL\benchmarks\google-benchmark\
build\third_party\googletest\build\googlemock\gmock_main.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest\internal\gtest-internal.h(625): warning C6326: Возможное сравнение константы с другой константой. [C:\Dev\STL\benchmarks\google-benchmark\b
uild\third_party\googletest\build\googlemock\gmock_main.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest/internal/gtest-internal.h(627,1): error C2220: следующее предупреждение рассматривается как ошибка [C:\Dev\STL\benchmarks\google-benchmark\
build\third_party\googletest\build\googlemock\gmock_main.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest\internal\gtest-internal.h(625): warning C6326: Возможное сравнение константы с другой константой. [C:\Dev\STL\benchmarks\google-benchmark\b
uild\third_party\googletest\build\googlemock\gmock_main.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest/internal/gtest-internal.h(627,1): error C2220: следующее предупреждение рассматривается как ошибка [C:\Dev\STL\benchmarks\google-benchmark\
build\third_party\googletest\build\googlemock\gmock_main.vcxproj]
C:\Dev\STL\benchmarks\google-benchmark\build\third_party\googletest\src\googletest\include\gtest\internal\gtest-internal.h(625): warning C6326: Возможное сравнение константы с другой константой. [C:\Dev\STL\benchmarks\google-benchmark\b
uild\third_party\googletest\build\googlemock\gmock_main.vcxproj]
  Создание кода...

```

When I enabled it by adding `add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/analyze:autolog->")` to CMakeLists.txt
So maybe we will add MSVS code analyzer in another PR, won't we?

LebedevRI commented:
> When I enabled it by adding `add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/analyze:autolog->")` to CMakeLists.txt So maybe we will add MSVS code analyzer in another PR, won't we?

Absolutely. In fact, i was only asking about the warnings part, not the static analysis, but thank you for checking!

LebedevRI commented:
@fsb4000 thank you!

fsb4000 commented:
@LebedevRI  My pleasure!
